docs/concepts/metrics.md

```diff
@@ -0,0 +1,85 @@
+| workflow_task_execution_failed            | A workflow task execution failed                                                                                                                                | Counter     | Core, Go, Java                              | namespace, task_queue, workflow_type                |
+| workflow_task_execution_latency           | Workflow task execution time in milliseconds                                                                                                                    | Histogram   | Core, Go, Java                              | namespace, task_queue, workflow_type                |
+| workflow_task_queue_poll_empty            | A workflow task queue poll timed out / had empty response                                                                                                       | Counter     | Core, Go, Java                              | namespace, task_queue, workflow_type                |
+| workflow_task_queue_poll_succeed          | A workflow task queue poll succeeded returning a new WFT                                                                                                        | Counter     | Core, Go, Java                              | namespace, task_queue, workflow_type                |
+| workflow_task_replay_latency              | Time it takes to catch up on replaying a WFT in ms                                                                                                              | Histogram   | Core, Go, Java                              | namespace, task_queue, workflow_type                |
+| workflow_task_schedule_to_start_latency   | Workflow task schedule-to-start time in milliseconds                                                                                                            | Histogram   | Core, Go, Java                              | namespace, task_queue, workflow_type                |
+
+#### Code definitions
+
+If you're curious about how the metrics are defined in source, see:
+
+- Core - [SDK metrics](https://github.com/temporalio/sdk-core/blob/master/core/src/telemetry/metrics.rs) and [Client metrics](https://github.com/temporalio/sdk-core/blob/master/client/src/metrics.rs)
+- Java - [SDK metrics](https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/internal/metrics/MetricsType.java) and [Client metrics](https://github.com/temporalio/sdk-java/blob/master/temporal-serviceclient/src/main/java/io/temporal/serviceclient/MetricsType.java)
+- Go - [SDK & Client metrics](https://github.com/temporalio/sdk-go/blob/c32b04729cc7691f80c16f80eed7f323ee5ce24f/internal/common/metrics/constants.go)
+
+### Notes on differences between SDKs
+
+- Histograms may have different buckets in different languages.
+- Some labels may not apply in every SDK
+
+## Server Metrics
+
+Documentation coming soon
```

Comment:
I think we have some docs about server side metrics.
@flossypurse?

Reply:
Can we aim to put those into a separate file?
i.e. Cluster metrics reference ?

Reply:
We do have this: https://docs.temporal.io/docs/server/production-deployment#scaling-and-metrics 
But it is not in a reference format or complete.